cpp/src/arrow/util/io_util.h

```diff
@@ -346,5 +346,11 @@ int64_t GetRandomSeed();
 ARROW_EXPORT
 uint64_t GetThreadId();
 
+/// \brief Get the current memory used by current process in bytes
+///
+/// This function support Windows and Linux
```

Comment:
```suggestion
/// \brief Get the current memory used by the current process in bytes
///
/// This function supports Windows, Linux, and Mac and will return 0 otherwise
```